src/containers/BooksForm.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { addBook } from '../actions/index';
+
+const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: '',
+      title: '',
+      category: 'Action',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({ id: Math.floor(Math.random() * 1000), [e.target.name]: e.target.value });
+  }
+
+  handleSubmit(e) {
+    const { title } = this.state;
+    if (title !== '') {
+      e.preventDefault();
+      const { addBook } = this.props;
+      addBook(this.state);
+      this.setState({
+        id: '',
+        title: '',
+        category: 'Action',
+      });
+      e.target.reset();
+    } else {
+      e.preventDefault();
+      alert('Please Type a valid Book Title'); // eslint-disable-line
+    }
```

Comment:
- [ ] There are needless repetitions here. You want to generally avoid long nested if conditionals. So you can simply do this instead:
```
if(!title) {
// return show error
}
e.preventDefault();
// rest of the code here...
```